### PR TITLE
Inline ops that read settings

### PR DESCRIPTION
**Pull request by miguelff**
Creating the patch first against the branch we use in prisma-engines.

### CODE REVIEW

**Comment by miguelff** [comment]
Screenshot 2023-04-19 at 17 03 04

Measures taken on 100 coldstart connections:

Before patch:
Average: 222.01605959
Standard deviation: 20.28249939282523

After patch:

Average: 152.8317912
Standard deviation: 22.382571070107467

Inline queries makes connection  run ~30% faster.

File: src/conn/mod.rs [hunk 1]
```diff
@@ -882,38 +880,51 @@ impl Conn {
...
-        Ok(())
-    }
 
-    /// Reads and stores `max_allowed_packet` in the connection.
-    async fn read_max_allowed_packet(&mut self) -> Result<()> {
-        let max_allowed_packet = if let Some(value) = self.opts().max_allowed_packet() {
-            Some(value)
+        // set max_allowed_packet
+        let max_allowed_packet = if read_max_allowed_packet {
+            settings.as_ref().map(|s| s.1).unwrap()
```
Comment (Weakky) [inline src/conn/mod.rs]: Nitpick: you _could_ get the result as a `Row` and then use `row::get("column_name")`. Might increase visibility, nothing blocking anyway.

**Review by Weakky — APPROVED** [review]
Nice! 👍

File: src/conn/mod.rs (hunk 1 above)
Comment (miguelff) [reply src/conn/mod.rs]: Good feedback, addressing.

**Comment by miguelff** [comment]
Cool, did a pass to the test suite using the same configuration defined in `azure-pipeliens.yml`. With MariaDB 10.5. Tests passed. When removing the line that reads the settings a few tests break. Tests indeed cover reading the settings.

**Comment by miguelff** [comment]
Measures taken on 100 coldstart connections to a remote RDS mysql 5.7 in us-east-1, accessed from Spain. Units are ms.

Before patch:

Average: 222.01605959
Standard deviation: 20.28249939282523

After patch:

Average: 152.8317912
Standard deviation: 22.382571070107467

Inlining queries make connections that require reading server settings run ~30% faster.

**Pull request merged**